src/bsp/include/bsp/TCPClientWrapper.h

```diff
@@ -0,0 +1,30 @@
+#ifndef __TCPClientWrapperWRAPPER_H_
+#define __TCPClientWrapperWRAPPER_H_
+
+#include "bsp/ITCPClient.h"
+#include <logger/ILogger.h>
+
+/**
+ *@brief A TCP client wrapper that is as a common socket between platforms.
+ **/
+class TCPClientWrapper : public ITCPClient {
```

Comment:
What is the goal of the `TCPClientWrapper` considering it is only returns calls from the `TCPClient` directly? I know it's probably some obvious OOP way of doing things, but it's been a long time since I've used OOP...

Reply:
The  TCPClient is a platform dependent implementation, honestly that's the only reason, and I cannot return a ITCPClient here because optional is a template and does not support reference or incomplete types.